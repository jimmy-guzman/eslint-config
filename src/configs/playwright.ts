import { GLOB_PLAYWRIGHT } from "../constants";
import { playwrightRules } from "../rules/playwright";
import { interopDefault } from "../utils/interop-default";

export default async function playwrightConfig() {
  const playwrightPlugin = await interopDefault(
    import("eslint-plugin-playwright"),
  );

  return [
    {
      ...playwrightPlugin.configs["flat/recommended"],
      files: GLOB_PLAYWRIGHT,
      name: "jimmy.codes/playwright",
      rules: await playwrightRules(),
    },
  ];
}
