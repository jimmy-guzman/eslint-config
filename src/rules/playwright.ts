import type { Rules } from "../types";

import { interopDefault } from "../utils/interop-default";

export const playwrightRules = async () => {
  const playwrightPlugin = await interopDefault(
    import("eslint-plugin-playwright"),
  );

  return {
    ...playwrightPlugin.configs["flat/recommended"].rules,
    "playwright/expect-expect": "error",
    "playwright/max-nested-describe": "error",
    "playwright/no-conditional-expect": "error",
    "playwright/no-conditional-in-test": "error",
    "playwright/no-element-handle": "error",
    "playwright/no-eval": "error",
    "playwright/no-force-option": "error",
    "playwright/no-nested-step": "error",
    "playwright/no-page-pause": "error",
    "playwright/no-skipped-test": "error",
    "playwright/no-slowed-test": "error",
    "playwright/no-useless-await": "error",
    "playwright/no-useless-not": "error",
    "playwright/no-wait-for-selector": "error",
    "playwright/no-wait-for-timeout": "error",
    "playwright/valid-title": [
      "error",
      {
        mustMatch: {
          it: "^should",
        },
      },
    ],
  } satisfies Rules;
};
