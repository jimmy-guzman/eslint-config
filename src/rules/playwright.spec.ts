import { playwrightRules } from "./playwright";

test("should create playwright rules", async () => {
  await expect(playwrightRules()).resolves.toMatchSnapshot();
});
