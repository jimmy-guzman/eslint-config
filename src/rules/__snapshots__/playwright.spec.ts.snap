// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should create playwright rules 1`] = `
{
  "no-empty-pattern": "off",
  "playwright/expect-expect": "error",
  "playwright/max-nested-describe": "error",
  "playwright/missing-playwright-await": "error",
  "playwright/no-conditional-expect": "error",
  "playwright/no-conditional-in-test": "error",
  "playwright/no-element-handle": "error",
  "playwright/no-eval": "error",
  "playwright/no-focused-test": "error",
  "playwright/no-force-option": "error",
  "playwright/no-nested-step": "error",
  "playwright/no-networkidle": "error",
  "playwright/no-page-pause": "error",
  "playwright/no-skipped-test": "error",
  "playwright/no-slowed-test": "error",
  "playwright/no-standalone-expect": "error",
  "playwright/no-unsafe-references": "error",
  "playwright/no-useless-await": "error",
  "playwright/no-useless-not": "error",
  "playwright/no-wait-for-selector": "error",
  "playwright/no-wait-for-timeout": "error",
  "playwright/prefer-web-first-assertions": "error",
  "playwright/valid-describe-callback": "error",
  "playwright/valid-expect": "error",
  "playwright/valid-expect-in-promise": "error",
  "playwright/valid-title": "off",
}
`;
